Lib/test/test_threading.py

```diff
@@ -722,7 +730,80 @@ def func():
         _, out, err = assert_python_ok("-c", code)
         data = out.decode().replace('\r', '')
         self.assertEqual(err.decode('utf-8'), "")
-        self.assertEqual(data, "Thread-1 (func)\nTrue\nTrue\n")
+        self.assertEqual(data,
+                         "Thread-1 (func) Thread\n"
+                         "True\n"
+                         "True\n"
+                         "True\n")
+
+    @unittest.skipIf(sys.platform in platforms_to_skip, "due to known OS bug")
+    @support.requires_fork()
+    @unittest.skipUnless(hasattr(os, 'waitpid'), "test needs os.waitpid()")
+    def test_main_thread_after_fork_from_foreign_thread(self, create_dummy=False):
+        code = """if 1:
+            import os, threading, sys, traceback, _thread
+            from test import support
+
+            def func(lock):
+                ident = threading.get_ident()
+                if %s:
+                    # call current_thread() before fork to allocate DummyThread
+                    current = threading.current_thread()
+                    print(current.name, type(current).__name__)
+                print(ident in threading._active)
+                # flush before fork, so child won't flush it again
+                sys.stdout.flush()
+                pid = os.fork()
+                if pid == 0:
+                    main = threading.main_thread()
+                    print(main.name, type(main).__name__)
+                    print(main.ident == ident)
+                    print(main is threading.current_thread())
+                    print(main.ident == threading.get_ident())
```

Comment:
You may add prefix for readability:

```suggestion
                    print("name", main.name, type(main).__name__)
                    print("is ident", main.ident == ident)
                    print("is current_thread", main is threading.current_thread())
                    print("get_ident", main.ident == threading.get_ident())
```